cdcprogresspb: move ResolvedTables proto into new package

andyyang890 opened the pull request:
This commit moves the `ResolvedTables` protobuf from `changefeedpb` to a new
`cdcprogresspb` package so that it can be added to the processor specs.
This was previously not possible because `pkg/sql/execinfrapb` (where the
processor specs live) disallows imports of `pkg/sql/parser`, which is a
dependency of a different protobuf in `changefeedpb`.

The `ProtectedTimestampRecords` protobuf has also been moved for consistency.

Informs #148119

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152883)
<!-- Reviewable:end -->

andyyang890 committed:
cdcprogresspb: move ResolvedTables proto into new package

This commit moves the `ResolvedTables` protobuf from `changefeedpb` to a new
`cdcprogresspb` package so that it can be added to the processor specs.
This was previously not possible because `pkg/sql/execinfrapb` (where the
processor specs live) disallows imports of `pkg/sql/parser`, which is a
dependency of a different protobuf in `changefeedpb`.

The `ProtectedTimestampRecords` protobuf has also been moved for consistency.

Release note: None

andyyang890 commented:
TFTRs!

bors r=aerfrei,asg0451
